core-bundle/src/Image/ImageSizes.php

```diff
@@ -107,17 +107,32 @@ private function loadOptions(): void
         $this->options = $GLOBALS['TL_CROP'];
 
         $rows = $this->connection->fetchAll(
-            'SELECT id, name, width, height FROM tl_image_size ORDER BY pid, name'
+            'SELECT
+                s.id, s.name, s.width, s.height, t.name as theme
+            FROM
+                tl_image_size s
+            LEFT JOIN
+                tl_theme t ON s.pid=t.id
+            ORDER BY
+                s.pid, s.name'
         );
 
+        $options = [];
+
         foreach ($rows as $imageSize) {
-            $this->options['image_sizes'][$imageSize['id']] = sprintf(
```

Comment:
I think we should add something like this here:

```php
if (\in_array($imageSize['theme'], ['exact', 'relative', 'image_sizes'], true)) {
    $imageSize['theme'] = 'Theme '.$imageSize['theme'];
}
```

Because if a theme has one of those three names the image size widget fails.

Reply:
Would it not be better to reserve these names for themes?

Reply:
I don’t think reserving is necessary. This is only about the labels for the `<optgroup>`s

Reply:
Implemented in 74d42cb4bfdd4d3aca81c1c3c31e8796834e9160.